### PR TITLE
add clear-cache.js

### PR DESCRIPTION
**Pull request by novain1102**
Re: #https://github.com/cocos/3d-tasks/issues/16194

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: scripts/clear-cache.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            }
+          });
+          fs.rmdirSync(dirPath);
+          console.log("正删除 "+ dirPath);
+    } else {
+        fs.unlinkSync(dirPath);
+        console.log("正删除 "+ dirPath);
+    }
+  } else {
+    console.log("不存在 " + dirPath);
```
Comment (PPpro) [inline scripts/clear-cache.js]: please use English in engine repo

File: scripts/clear-cache.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    path.join(__dirname, "../DebugInfos.json"),
+    path.join(__dirname, "../bin/"),
+    path.join(__dirname, "../platforms/runtime/local-commit.json"),
+    path.join(__dirname, "../scripts/native-pack-tool/dist/"),
+    path.join(__dirname, "../scripts/typedoc-plugin/lib/"),
+    path.join(__dirname, "../native/simulator/"),
+    path.join(__dirname, "../native/tools/.DS_Store"),
+    path.join(__dirname, "../native/build/"),
+    path.join(__dirname, "../native/compile_commands.json"),
+    path.join(__dirname, "../native/.cache/"),
```
Comment (PPpro) [inline scripts/clear-cache.js]: Do some of these files comes from `.gitignore` file ?

can we dynamically get the files/dir from this file.

Comment (novain1102) [reply scripts/clear-cache.js]: what do you mean `dynamically get the files/dir from this file`

Comment (novain1102) [reply scripts/clear-cache.js]: > Do some of these files comes from .gitignore file ?

just some files delete by git clean -xfd without  `node_module` related

Comment (PPpro) [reply scripts/clear-cache.js]: I mean this array of files should comes from `.gitignore`, 
now the issue of current implementation is that someday if I update `.gitignore`, then I should always remember to update this array of files to delete.

Comment (novain1102) [reply scripts/clear-cache.js]: done.

**Review by PPpro — APPROVED** [review]

File: scripts/clear-cache.js (hunk 1 above)
Comment (novain1102) [reply scripts/clear-cache.js]: done.

**Review by VisualSJ — APPROVED** [review]

**Pull request merged**